api/envoy/config/core/v3/base.proto

```diff
@@ -444,3 +444,12 @@ message ControlPlane {
   // the Envoy is connected to.
   string identifier = 1;
 }
+
+// DNS resolver configuration which includes the underlying dns resolver addresses and options
```

Comment:
Nit: suggest moving this to its own file `resolver.proto` in `api/envoy/config/v3`.

Reply:
While we are at this, what does others think about renaming `DnsResolver` to `DnsResolverConfig` ? So that we don't get confused between this message and a class of the same name [DnsResolver](https://github.com/envoyproxy/envoy/blob/main/include/envoy/network/dns.h#L44). Apologies for not bringing this up earlier.

Reply:
Sure, although I think that's a good idea just because the naming scheme is clearer; avoiding C++ level type conflicts isn't that strong a motivator, we have namespaces etc to deal with that.

Reply:
is it also possible to foward ip directly without dns query?

I did curl -x http://envoyip:port -kLv http://targetip:port but envoy keep returning "No healthy upstream"

Reply:
So, definitely open to (maybe it's clearer) to call this `DnsResolverConfig` or `DnsResolutionConfig`.

Reply:
let's say, we can take that up renaming `DnsResolver` --> `DnsResolverConfig` or `DnsResolutionConfig` in the next PR #16237 ? Since all the CI got passed here, this can go?